go/git/put.go

```diff
@@ -0,0 +1,57 @@
+package git
+
+import (
+	"context"
+	"encoding/base64"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-codec/codec"
+)
+
+func PutMetadata(ctx context.Context, g *libkb.GlobalContext, arg keybase1.PutGitMetadataArg) error {
+	// TODO: replace these with real implementations
+	var teamer Teamer
+	cryptoer := NewCrypto(g)
+
+	teamIDVis, err := teamer.LookupOrCreate(ctx, arg.Folder)
+	if err != nil {
+		return err
+	}
+
+	// Translate the GitLocalMetadata struct into GitLocalMetadataVersioned,
+	// for versioned storage.
+	localMetadataVersioned := keybase1.NewGitLocalMetadataVersionedWithV1(
+		keybase1.GitLocalMetadataV1{
+			RepoName: arg.Metadata.RepoName})
+
+	mh := codec.MsgpackHandle{WriteExt: true}
+	var msgpackLocalMetadata []byte
+	enc := codec.NewEncoderBytes(&msgpackLocalMetadata, &mh)
+	err = enc.Encode(localMetadataVersioned)
+	if err != nil {
+		return err
+	}
+	encryptedMetadata, err := cryptoer.Box(ctx, msgpackLocalMetadata, teamIDVis)
+	if err != nil {
+		return err
+	}
+	base64Ciphertext := base64.StdEncoding.EncodeToString(encryptedMetadata.E)
+	base64Nonce := base64.StdEncoding.EncodeToString(encryptedMetadata.N[:])
+
+	apiArg := libkb.APIArg{
+		Endpoint:    "kbfs/git/team/put",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+		Args: libkb.HTTPArgs{
+			"team_id":            libkb.S{Val: string(teamIDVis.TeamID)},
+			"repo_id":            libkb.S{Val: string(arg.RepoID)},
+			"encrypted_metadata": libkb.S{Val: base64Ciphertext},
+			"nonce":              libkb.S{Val: base64Nonce},
+			"key_generation":     libkb.I{Val: int(encryptedMetadata.Gen)},
+			"device_id":          libkb.S{Val: string(g.Env.GetDeviceID())},
+		},
+	}
+	_, err = g.GetAPI().Post(apiArg)
```

Comment:
encryptedMetadata.V is dropped?

Reply:
Currently the server stores (ciphertext, nonce, key gen) in 3 separate fields. I could add version to that, or I could serialize the `EncryptedGitMetadata` struct and shove the whole thing in a single field. Preference?

Reply:
The background here is that @patrickxb introduced `V` after I'd already created the DB tables, and I haven't caught up yet.

Reply:
Ah ok. No preference. Seems like if this had happened in the other order you would have added a new field.

Reply:
I kind of liked splitting the fields up, because it let me add a foreign key on the key generation, so that at least if you tried to upload a blob claiming to be encrypted for a key that didn't exist, we wouldn't allow it. But admittedly, that's probably not going to happen very often. @patrickxb any preference?

Reply:
I think it would be nice as separate fields as opposed to something serialized.